core-rust/mesh-api-server/src/mesh_api/handlers/account_balance.rs

```diff
@@ -0,0 +1,111 @@
+use crate::prelude::*;
+
+pub(crate) async fn handle_account_balance(
+    state: State<MeshApiState>,
+    Json(request): Json<models::AccountBalanceRequest>,
+) -> Result<Json<models::AccountBalanceResponse>, ResponseError> {
+    // TODO assert sub_network
+    assert_matching_network(&request.network_identifier.network, &state.network)?;
+    assert_account(&request.account_identifier)?;
+
+    let mapping_context = MappingContext::new(&state.network);
+    let extraction_context = ExtractionContext::new(&state.network);
+
+    let component_address =
+        extract_component_address(&extraction_context, &request.account_identifier.address)
+            .map_err(|err| err.into_response_error("account_identifier.address"))?;
+
+    let database = state.state_manager.database.snapshot();
+
+    let header = if let Some(_) = request.block_identifier {
+        return Err(ResponseError::from(ApiError::InvalidRequest)
+            .with_details("Historical balance not supported"));
+    } else {
+        read_current_ledger_header(database.deref())
+    };
+
+    let type_info: Option<TypeInfoSubstate> = read_optional_substate::<TypeInfoSubstate>(
+        database.deref(),
+        component_address.as_node_id(),
+        TYPE_INFO_FIELD_PARTITION,
+        &TypeInfoField::TypeInfo.into(),
+    );
+
+    if type_info.is_none() {
+        if component_address.as_node_id().is_global_preallocated() {
```

Comment:
Do we need to support identity as well? How about just `EntityType::GlobalPreallocatedSecp256k1Account | EntityType::GlobalPreallocatedEd25519Account`?

Reply:
Added `extract_account()` which checks for `GlobalAccount | GlobalPreallocatedSecp256k1Account GlobalPreallocatedEd25519Account`